virtwho/virt/virt.py

```diff
@@ -316,9 +316,47 @@ def hash(self):
         return hashlib.sha256(json.dumps(self.serializedAssociation, sort_keys=True).encode('utf-8')).hexdigest()
 
 
+class StatusReport(AbstractVirtReport):
+    """
+    Status Report from virt backend about ability to connect.
+    """
+
+    def __init__(self, config, state=AbstractVirtReport.STATE_CREATED):
+        super(StatusReport, self).__init__(config, state)
+        self._source_message = ''
+        self._destination_message = ''
+        self._destination = ''
+
+    @property
+    def data(self):
+        # make the components into dictionary
+        data = {"source":{"server": self.config['server'] if 'server' in self.config else 'none',
+                "status_string": 'success' if len(self._source_message) == 0 else 'failure'},
+                "destination":{"server": self._destination,
+                "status_string": 'success' if len(self._destination_message) == 0 else 'failure'}}
+        if len(self._source_message) > 0:
+            data['source']['message'] = self._source_message
+        if len(self._destination_message) > 0:
+            data['destination']['message'] = self._destination_message
```

Comment:
It is interesting that you mix two approaches. One big assignment on line 333 and then you extend the dictionary here. The question is why? I miss probably miss something.

Reply:
I didn't want the element label to appear if there is no message, but it's not a big deal. updated.